src/plugins/intel_gpu/src/plugin/ops/select.cpp

```diff
@@ -33,54 +33,54 @@ static void CreateSelectOp(Program& p, const std::shared_ptr<ngraph::op::v1::Sel
         // Preprocess inputs
         for (size_t i = 0; i < inputPrimitives.size(); ++i) {
             auto input_pshape = op->get_input_partial_shape(i);
-            OPENVINO_ASSERT(input_pshape.is_static(), "Dynamic shapes are not supported for v1::Select with NUMPY mode yet");
-            auto input_shape = input_pshape.to_shape();
-            auto input_rank = input_shape.size();
-
-            // Add reorder if changing number of dimensions requires changing format
-            auto targetFormat = cldnn::format::get_default_format(output_rank);
-
-            if (targetFormat.value != cldnn::format::get_default_format(input_rank).value) {
-                auto reorderName = layerName + "_cldnn_in" + std::to_string(i) + "_reorder";
-                auto targetDatatype = cldnn::element_type_to_data_type(op->get_input_element_type(i));
-                auto reorderPrim = cldnn::reorder(reorderName,
-                                                  inputPrimitives[i],
-                                                  targetFormat,
-                                                  targetDatatype,
-                                                  std::vector<float>(),
-                                                  cldnn::reorder_mean_mode::subtract);
-
-                p.add_primitive(*op, reorderPrim);
-
-                inputPrimitives[i] = reorderName;
-            }
 
-            // Reshape input if they differ or select specific shape matches default one
-            if (input_rank != output_rank || input_rank < 4) {
-                auto reshapeName = layerName + "_cldnn_in" + std::to_string(i) + "_reshape";
+            if (input_pshape.is_static()) {
```

Comment:
Even though input is static, if we are using the new shape infer in the network, we shoulndt add reorders. So please change it to !p.use_new_shape_infer()

Reply:
done